[AP Port] Lamias

PoltavskaPraca opened the pull request:
## About The Pull Request
 **Need some more checks and naturally TESTMERGE but it works proper and fine (from my testing)**
                                           port of https://github.com/GeneralPantsuIsBadAtCoding/Azure-Peak/pull/3224

+ Adds lamias as a separate race. Functions as both half-kin and a full snek race? Don't feel the reason to separate them, to avoid the bloat. 
+ Avaialable to pick skin colors from the wheel (or a hex colour from the hexagonal thing-diagram??? idk) AND predefined skintones (elf + goblin). Little (?) tooltip with a skin tone hex code reference sheet.
+ Tail HP is 300.
+ Right now they can wear both boots and the leg armour, no restrictions. However, the sprites for those aren't rendered. Examine text for boots says 'they are wearing X on the tip of the tail'.
+ Innate racial longstrider, strong stomach (eat raw food + murky water) and +1 ST, -1 SPD. No SPD clamping.
+ Reviled strata
+ Lamias use custom racial sprites like dwarf/goblins/etc. So if something doesn't render? Needs the racial variant in .dmi
+ ADJUSTED EVERY FUCKING SPRITE THAT GOES INTO SHIRT/ARMOR/CLOAK SLOT. HOLY SHIT.
+ Innate auto-hissing via the tongue organ, no accent.
+ Kicks that displace people by TWO tiles,  enabling knock overs from +1 tile away.
+ Trait-based verbage change, replacing kick and stomp with relevant verbs.
+ Unique footstep sfx.
+ sprites todo:
custom leg (tail) armour
3 markings sprites for the tail mirroring those of snakes
tabards.dmi
shirts.dmi
shirts_gown.dmi
shirts_royal.dmi?
stalker robe sides

+ to do:
new leg (tail) armour + craft recipes
adjust limb hp?
inspect RIGHT LEG
+1 swimming roundstart ??
tail limb hp?
roundendstats look up axian

## Testing Evidence
<img width="416" height="343" alt="image" src="https://github.com/user-attachments/assets/553ed3bc-edf7-4e7c-b44e-6a9a46c5b046" />
<img width="696" height="656" alt="image" src="https://github.com/user-attachments/assets/8aaaef2f-d249-47e8-809f-5b12b56df1fa" />
<img width="629" height="749" alt="image" src="https://github.com/user-attachments/assets/fc61056d-dc41-4581-b987-c53c174985c4" />
<img width="498" height="268" alt="image" src="https://github.com/user-attachments/assets/1cdb2861-f784-44fa-9b4b-d926a32efe7b" />
<img width="532" height="480" alt="image" src="https://github.com/user-attachments/assets/f82f5ec5-424c-4e70-9114-0edde0caab3c" />

## Why It's Good For The Game

<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/231839e3-a65d-4756-bf96-46925cf11761" />

PoltavskaPraca commented:
IDK how to move it to draft instead of open.

sippykot commented:
> currenlty they can't wear boots, therefore they don't have any armour on their feet whatsoever. suggest giving them back their boots
> 
> +1st -1 spd, speed clamped in 10-12 range, innate mudwalking or just either +1 spd or +1st

snake shoes? that would make 0 sense

Mannybrado commented:
doesnt AP give their legs innate armor

ShadowDeath6 commented:
so are they wildkin or are they their own species?

coffeecoffeecoffee commented:
These are INCREDIBLY awesome. I love this. However, having seen the way the dullahan PR has been received, I feel like we absolutely need to do _something_ to encourage these to be played as _monsters_ and be shunned rather than accepted as normal adventurers.

PoltavskaPraca commented:
> > currenlty they can't wear boots, therefore they don't have any armour on their feet whatsoever. suggest giving them back their boots
> > +1st -1 spd, speed clamped in 10-12 range, innate mudwalking or just either +1 spd or +1st
> 
> snake shoes? that would make 0 sense

Sprite?

PoltavskaPraca commented:
> These are INCREDIBLY awesome. I love this. However, having seen the way the dullahan PR has been received, I feel like we absolutely need to do _something_ to encourage these to be played as _monsters_ and be shunned rather than accepted as normal adventurers.

The devious dog/fox/sparklefur that walks upright:

Idk though, they could get the same jobs as the half-orcs. Only real reason to shun them, I believe, is because they are fucking heavy, slithering and DO look imposing.

On the lore-side of things, I want them to be Abyssor-adjacent bc fuck Dendor and he isn't the only one who should give his patronage to non-human.

PoltavskaPraca commented:
Almost forgot, but they have the fattest sprite possible, meaning, it's even easier to actually click on them for an attack.

PoltavskaPraca commented:
> so are they wildkin or are they their own species?

They are their own species, entirely.



> I like them being abyssor-adjacent, sea serpents are cool mythos things you could give them like snake / serpent skin colors perhaps, like dark green / blue

I am not give them snake/serpenet skin colours. It's all up to the players, they have a whole HEX-palette to them.

coffeecoffeecoffee commented:
> > These are INCREDIBLY awesome. I love this. However, having seen the way the dullahan PR has been received, I feel like we absolutely need to do _something_ to encourage these to be played as _monsters_ and be shunned rather than accepted as normal adventurers.
> 
> The devious dog/fox/sparklefur that walks upright:
> 
> Idk though, they could get the same jobs as the half-orcs. Only real reason to shun them, I believe, is because they are fucking heavy, slithering and DO look imposing. Could have a long-history of coastal raiding and whatnot, tribal, too.
> 
> On the lore-side of things, I want them to be Abyssor-adjacent bc fuck Dendor and he isn't the only one who should give his patronage to non-human.
> 
> As for the mechanics... Innate longstrider, 1 str -1 spd but their speed is always clamped to be in-between 9-12. Temperature, weight and other things are not implement here whatsoever and it def would be a no-brainer to make them dependant on ambient temperature. Another mechanic would be letting them grab people but instead of taking up the hand slot, the grab would be on their feet slot or whatever. How are the people going to hotkey dropping the grab? No idea, unless I want to add so too.

Absolutely. Maybe they could be restricted to things like vagabond and wretch, though? You **_know_** people are absolutely still going to play them even if the racelocks on them are super restrictive, but I thoroughly believe these should be implemented as **monster** characters.

ShadowDeath6 commented:
> I am not give them snake/serpenet skin colours. It's all up to the players, they have a whole HEX-palette to them.

😔 it's over.

coffeecoffeecoffee commented:
> > I am not give them snake/serpenet skin colours. It's all up to the players, they have a whole HEX-palette to them.
> 
> 😔 it's over.

i think this means you can just _pick_ snake/serpent/reptile skin colors because you get the whole color wheel?

PoltavskaPraca commented:
> > I am not give them snake/serpenet skin colours. It's all up to the players, they have a whole HEX-palette to them.
> 
> 😔 it's over.

this is your tail colour selection
<img width="531" height="389" alt="image" src="https://github.com/user-attachments/assets/33f615d4-65d3-46cb-90e4-92d601836a2e" />

this is your skin/detail/whatever colour selection
<img width="506" height="351" alt="image" src="https://github.com/user-attachments/assets/89252898-9071-4a05-8184-df6c289ef01a" />

PoltavskaPraca commented:
### [a416db0](https://github.com/Scarlet-Reach/Scarlet-Reach/pull/509/commits/a416db0cab0a8fb8d38ab9e912a06ed6d05f05ef) changelog

+ Added lore?
The monstrous spawn of Abyssor, snake and humen conjoined together. While one could consider them to be of Dendor's, he had no hand in their creation. Lamia are widespread in the southern coastal regions, where their tribes have settled in aeons ago, much of their written and oral history is filled with accounts	of grand raids on coastal regions, for they have been terrorizing any race that has dared to settle near their waters. For this, they are widely shunned by the other races,	with the exception of Axians with whom they share their natural heartlands. Many a sailor has met their end at the claws of Lamias.

+ Stat bonuses/maluses removed. Instead, innate longstrider trait + raw food trait.
+ Gave them back their fucking shoes. Without any foot armour, they immediately proceed to get shit on. Imagine wearing full plate but your pinky is exposed so you just kinda die? Yeah, this is lamia without shoes. **Shoes and pants still do not render on the sprite.**
+ Introduced SPEED stat clamping to Lamias. Lamia speed stat cannot go lower than 9 or higher than 12. Any buff/debuff that would increase/decrease their speed is not going to do a thing.
+ Shunned race, this is what role they CAN be in the round.
<img width="1156" height="611" alt="Screenshot 2025-08-10 163517" src="https://github.com/user-attachments/assets/9051db6a-0818-437b-a0ad-075c85a3d6d4" />

ShadowDeath6 commented:
> * Gave them back their fucking shoes. Without any foot armour, they immediately proceed to get shit on. Imagine wearing full plate but your pinky is exposed so you just kinda die? Yeah, this is lamia without shoes

Would it be possible to give them skin_armor on legs and feet like werewolves and wildshapes have? Might be the easiest option if you don't want to go about spriting unique armors for them.

PoltavskaPraca commented:
> > * Gave them back their fucking shoes. Without any foot armour, they immediately proceed to get shit on. Imagine wearing full plate but your pinky is exposed so you just kinda die? Yeah, this is lamia without shoes
> 
> Would it be possible to give them skin_armor on legs and feet like werewolves and wildshapes have? Might be the easiest option if you don't want to go about spriting unique armors for them.

I'd rather not give them skin armour bc it's fucking lame and seeing how it works on werewolves it's probably going to be a pain in the ass to fix. How is it going to regenerate? Peeling? Etc etc.

coffeecoffeecoffee commented:
> > > * Gave them back their fucking shoes. Without any foot armour, they immediately proceed to get shit on. Imagine wearing full plate but your pinky is exposed so you just kinda die? Yeah, this is lamia without shoes
> > 
> > 
> > Would it be possible to give them skin_armor on legs and feet like werewolves and wildshapes have? Might be the easiest option if you don't want to go about spriting unique armors for them.
> 
> I'd rather not give them skin armour bc it's fucking lame and seeing how it works on werewolves it's probably going to be a pain in the ass to fix. How is it going to regenerate? Peeling? Etc etc.

I think they meant only for their legs/feet.
Either way I feel like lamias should probably have some kind of restriction to only the "skirt" and "dress" items, if you can swing it. There are armored kilts and the like in the game available for these purposes I think.
<img width="158" height="120" alt="dreamseeker_CtmA2HSIUg" src="https://github.com/user-attachments/assets/49f783fe-87e8-4655-8b86-32ca1e7bc6b5" />
Actually, these *also* protect the feet... Hmm...

PoltavskaPraca commented:
> > > These are INCREDIBLY awesome. I love this. However, having seen the way the dullahan PR has been received, I feel like we absolutely need to do _something_ to encourage these to be played as _monsters_ and be shunned rather than accepted as normal adventurers.
> > 
> > 
> > The devious dog/fox/sparklefur that walks upright:
> > Idk though, they could get the same jobs as the half-orcs. Only real reason to shun them, I believe, is because they are fucking heavy, slithering and DO look imposing. Could have a long-history of coastal raiding and whatnot, tribal, too.
> > On the lore-side of things, I want them to be Abyssor-adjacent bc fuck Dendor and he isn't the only one who should give his patronage to non-human.
> > As for the mechanics... Innate longstrider, 1 str -1 spd but their speed is always clamped to be in-between 9-12. Temperature, weight and other things are not implement here whatsoever and it def would be a no-brainer to make them dependant on ambient temperature. Another mechanic would be letting them grab people but instead of taking up the hand slot, the grab would be on their feet slot or whatever. How are the people going to hotkey dropping the grab? No idea, unless I want to add so too.
> 
> A while back I had started drafting up some ideas on how to implement a naga race that would be Abyssor-adjacent (cool to see another person had that idea as well) and made some mockup sprites to how they would look. I think a really good way to encourage these being played as monstrous beings rather than giant puppydogs (nothing wrong with that but come on we grimshart in this bitch) is to give them some pretty scary stats and traits while limiting them to roles that put them at a disadvantage with the keep or at odds with the keep (and the church, naturally).
> 
> edit: Of course, I made my sprites in 42x42 because I wanted them to really stand out too, heh.

Would you mind sharing the sprites? Currently sprites for markings are very much required for customization.

PoltavskaPraca commented:
### aadb134

finally finished adjusting sprites from armor.dmi and added relevant entries in wrists.dmi

Genevesh commented:
This is amazing work right here. Keep it up, we're all rooting for you!
I think even the host who is reluctant to accept taurs shouldn't mind lamias, they are in more of the neutral territory and even then are still considered monstrous enough to restrict heavily. It'll work out.

PoltavskaPraca commented:
Finished adjusting sprites. Need more adjustments for dresses and shirts east/west.

PoltavskaPraca commented:
> Do they things like pants/shoes disabled as slots? Seems like they would. And if so... do they get any natural scale armor as a passive to help make up for that?

Right now they can wear both boots and the leg armour, no restrictions. However, the sprites for those aren't rendered. Examine text for anything in the boot slot says 'they are wearing X on the tip of the tail'.
natural scale armor - snowflake balancing etc etc. Also plated top and... a naked tail? Not a good look.
Without boot armour they are gonna eat shit in combat. End of story.

> what are the chances of adding a snake tongue organ that converts "s" into "ss"

Soon :tm:

Genevesh commented:
> Do they things like pants/shoes disabled as slots? Seems like they would. And if so... do they get any natural scale armor as a passive to help make up for that?

I believe the author wants them to have their own leg/foot armor once he has the sprites from them; At that point he will restrict them from wearing regular ole shoes and pants.

PoltavskaPraca commented:
<img width="691" height="551" alt="Screenshot 2025-08-12 220005" src="https://github.com/user-attachments/assets/991319d9-1669-428c-8be6-9cc4748d9ad6" />
<img width="777" height="867" alt="Screenshot 2025-08-12 220036" src="https://github.com/user-attachments/assets/b26a78d3-2ed3-4d43-bce1-bfcc715dde26" />
<img width="728" height="676" alt="Screenshot 2025-08-12 220048" src="https://github.com/user-attachments/assets/9cc7ab83-4c99-4f2c-8f82-b75760abd1f3" />

PoltavskaPraca commented:
<img width="455" height="177" alt="image" src="https://github.com/user-attachments/assets/7fbeecbd-55d3-4043-944c-f433217cd4b9" />
<img width="434" height="104" alt="image" src="https://github.com/user-attachments/assets/051b8a4a-0c3a-4066-b251-c49042f81d1a" />

EvenInDeathIStillServe commented:
Resolve thy conflict and I can TM you.

Vultchurr commented:
Alright... First and foremost issue that I (personally, anyways) noticed is that lamias have access to way too many roles. Like, _way too many_. They're towners, they're yeomen, _acolytes_ and even magos thralls. This means that they're now everywhere in the town, contradictory to their lore which implies that they are a heavily shunned and hated species due to their abyssal nature.
<img width="1097" height="489" alt="racelock" src="https://github.com/user-attachments/assets/529e925e-50e6-4c99-a3c8-095de6525692" />

I feel like the latter point could be partially resolved mechanically by giving them traits that inflict negative statuses on non-lamias, similar to how the xenophobia and ugly traits work.

Second issue is they have no access to the cobra hoods. Zardmen and wild-kin can both use them, but not lamia. A simple oversight I'm sure, since they have their own feature type. 
<img width="589" height="690" alt="hoodoption" src="https://github.com/user-attachments/assets/28bac8c4-0dc3-4084-a323-75c447c33c0b" />

PoltavskaPraca commented:
> Alright... First and foremost issue that I (personally, anyways) noticed is that lamias have access to way too many roles. Like, _way too many_. They're towners, they're yeomen, _acolytes_ and even magos thralls. This means that they're now everywhere in the town, contradictory to their lore which implies that they are a heavily shunned and hated species due to their abyssal nature.
> 
> I feel like the latter point could be partially resolved mechanically by giving them traits that inflict negative statuses on non-lamias, similar to how the xenophobia and ugly traits work.
> 
> Second issue is they have no access to the cobra hoods. Zardmen and wild-kin can both use them, but not lamia. A simple oversight I'm sure, since they have their own feature type. 

Sure. Could you provide a potential role list which in your opinion would be fitting? Just take the screenshot and paint black over roles, just black rectangle out NEVER in ms paint. 

As for xenophobia... perhaps. As it now stands, ugly VIRTUE is a VIRTUE because it debuffs whoever looks at you. Matter of fact, it is undesirable to pick Beautiful virtue if you ever wanna frag, bc it buffs the oponent.

Vultchurr commented:
> > Alright... First and foremost issue that I (personally, anyways) noticed is that lamias have access to way too many roles. Like, _way too many_. They're towners, they're yeomen, _acolytes_ and even magos thralls. This means that they're now everywhere in the town, contradictory to their lore which implies that they are a heavily shunned and hated species due to their abyssal nature.
> > I feel like the latter point could be partially resolved mechanically by giving them traits that inflict negative statuses on non-lamias, similar to how the xenophobia and ugly traits work.
> > Second issue is they have no access to the cobra hoods. Zardmen and wild-kin can both use them, but not lamia. A simple oversight I'm sure, since they have their own feature type.
> 
> Sure. Could you provide a potential role list which in your opinion would be fitting? Just take the screenshot and paint black over roles, just black rectangle out NEVER in ms paint.
> 
> As for xenophobia... perhaps. As it now stands, ugly VIRTUE is a VIRTUE because it debuffs whoever looks at you. Matter of fact, it is undesirable to pick Beautiful virtue if you ever wanna frag, bc it buffs the oponent.

<img width="1097" height="489" alt="racelock" src="https://github.com/user-attachments/assets/1e8cdf99-ae66-489e-8f50-8f152540376a" />

I feel like it might be a bit contentious, but my thought process is that they should mostly relegated to outsider roles. As for the exceptions, I think they make sense because...

**Hand:** Literally handpicked by the Duke/Duchess, their reasoning is matter of roleplay here.
**Merchant:** Wealth begets status, and with enough coin (as well as offering a lucrative service!) I'd think it reasonable for a lamia to own property as a foreigner.
**Dungeoneer:** People almost exclusively play this role as sadistic torture machines with zero personality beyond acting as monstrous as possible, so perhaps the Duchy wouldn't be so opposed to hiring a monstrous creature to terrorize their captives and potential criminals.
**Court Agent:** Similar to Hand, except Handpicked by the Hand. I've never played a court agent and I'm honestly not really sure they do besides act as spies.

In regards to ugly being a combat debuff, I won't argue since it's a fair point. The idea I had behind giving them the trait was to give a mechanical reason for having other races despise them, and a debuff seemed like a good way of going at it. That said though, a monster-specific trait that only gives the debuff to non-monsters would be more fitting since otherwise you'd have these cutesy half-snake half-anime babes vomiting at the sight of each other.

PoltavskaPraca commented:
> I feel like it might be a bit contentious, but my thought process is that they should mostly relegated to outsider roles. As for the exceptions, I think they make sense because...

I think adding acolyte/churchling and apothecary to the list won't make it all too abysmal. They still don't have any property but do get atleast SOME towner roles. Like, my whole idea is that they aren't accepted in a place just because, unless...

They are an acolyte of abyssor's/pestra/etc. and aren't that bad as a person, so they were granted a place at the church.
They came to the town and are a damn good surgeon/healer, so? With some mistrust they were allowed to be employed by the esculap.
Same reasoning for magos thrall, you'd expect one to be a freak to wind up as magos/magos thrall.

Vultchurr commented:
That's a fair compromise. It at least gives me a reason to pick back up on my naga stuff, 'cuz now there's a reason to differentiate them from the lamias.

PoltavskaPraca commented:
https://cdn.discordapp.com/attachments/1389452066493169765/1406288334137065573/Desktop_2025.08.16_-_21.46.18.15.mp4?ex=68a1eb92&is=68a09a12&hm=744ca3df472070811e17b3dbf90b6fdfbb95c92ebdcf3344664cdc7e744079bd&

Bobinater commented:
PR BROKE EXAMINE, GOOD JOB

PoltavskaPraca commented:
> PR BROKE EXAMINE, GOOD JOB

do a code review

CandleJaxx commented:
can you add like. a harpy wings allowment for lamia? specifically bc i want to make a fucked up sea creechur with a nagatail 
<img width="151" height="92" alt="image" src="https://github.com/user-attachments/assets/164de5ec-e146-47cb-b67f-2377c3f0e30a" />

PoltavskaPraca commented:
>can you add like. a harpy wings allowment for lamia? specifically bc i want to make a fucked up sea creechur with a nagatail

sure

CandleJaxx commented:
> > > can you add like. a harpy wings allowment for lamia? specifically bc i want to make a fucked up sea creechur with a nagatail
> > 
> > 
> > sure also, snout selection might need expanding to allow for more customization. Is there anything you'd like to be added to it? a name of the option you'd like will do
> 
> nope, but thank you!! excited :)

wait, i take that back! horns! please and thank you :)
